ext/mini_racer_extension/mini_racer_extension.cc

```diff
@@ -205,9 +203,6 @@ static VALUE rb_platform_terminate(VALUE self) {
 }
 
 static void init_v8() {
-    // no need to wait for the lock if already initialized
-    if (current_platform != NULL) return;
```

Comment:
is this really a concern in any practical sense? every time we init a context we call this so this adds 1 lock acquisition to every context creation.

Reply:
This pattern is called double-checked locking, and this particular implementation is unsafe. In amd64 it's safe as long as the compiler doesn't do some reordering of the assignments, but to ensure that you'd need to add compiler fences anyway. You can see a summary and what a correct implementation would like here: https://preshing.com/20130930/double-checked-locking-is-fixed-in-cpp11/

That said, I doubt there is a lot of contention on this lock (which after the initialization is only held to check if the pointer is not null), so I didn't implement the double-checked locking.

Reply:
fair enough, thanks for explaining!